### PR TITLE
User Creation

### PR DESCRIPTION
**Pull request by Prolas**
The user creation works as follows:
* An admin creates a temporary user which will be stored in the tempUser collection. The id of the document is the email.
* When a user signs it checks whether an entry exists in the user collection. If not it checks if an entry exists in tempUser. If yes a new user is added to the user collection and the entry in tempUser is deleted. Otherwise the user cannot log in.
These checks are in the MainActivity. Is it the right place ?

An email address has been added to User and UserSchema

### CODE REVIEW

File: app/src/main/java/ch/epfl/skysync/MainActivity.kt [hunk 1]
```diff
@@ -12,26 +12,59 @@ import androidx.navigation.compose.rememberNavController
...
   private fun onSignInResult(result: FirebaseAuthUIAuthenticationResult) {
     if (result.resultCode == RESULT_OK) {
-      user.value = FirebaseAuth.getInstance().currentUser
-      SnackbarManager.showMessage("Successfully signed in")
-    } else {
-      SnackbarManager.showMessage("Authentication failed")
+      val incomingUser = FirebaseAuth.getInstance().currentUser!!
+      val email = incomingUser.email!!
+
+      runBlocking {
```
Comment (Plouc314) [inline app/src/main/java/ch/epfl/skysync/MainActivity.kt:40]: There should not be a `runBlocking` block, instead you should define a suspend function elsewhere and wrap it in `lifecycleScope.launch`

Comment (Prolas) [reply app/src/main/java/ch/epfl/skysync/MainActivity.kt:41]: I will change it

File: app/src/main/java/ch/epfl/skysync/database/schemas/TempUserSchema.kt [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+) : Schema<TempUser> {
+  override fun toModel(): TempUser {
+    return when (userRole!!) {
+      UserRole.PILOT ->
+          TempUser(
+              email = email!!,
+              userRole = userRole!!,
+              firstname = firstname!!,
+              lastname = lastname!!,
+              balloonQualification = balloonQualification!!)
```
[the remark this answers is not in the document]
Comment (Prolas) [reply app/src/main/java/ch/epfl/skysync/database/schemas/TempUserSchema.kt:24]: I will change it

File: app/src/main/java/ch/epfl/skysync/database/tables/TempUserTable.kt [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+   *
+   * @param email The email used as id
+   * @param item The temporary user to add to the database
+   */
+  suspend fun set(email: String, item: TempUser, onError: ((Exception) -> Unit)? = null) {
+    return withErrorCallback(onError) { db.setItem(path, email, TempUserSchema.fromModel(item)) }
+  }
+
+  companion object {
+    const val PATH = "tempUser"
```
[the remark this answers is not in the document]
Comment (Prolas) [reply app/src/main/java/ch/epfl/skysync/database/tables/TempUserTable.kt:21]: I will change it

**Review by Plouc314 — APPROVED** [review]
LGTM

**Pull request merged**